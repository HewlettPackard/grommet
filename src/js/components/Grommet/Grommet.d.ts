import * as React from 'react';
import Grommet from '../index';

export declare namespace GrommetTypes {
  type Direction = 'rtl' | 'ltr';
}

export interface GrommetProps extends Grommet.Props {
  dir: GrommetTypes.Direction;
  theme: {};
}

export class Grommet extends React.Component<GrommetProps, undefined> { }

export default Grommet;
