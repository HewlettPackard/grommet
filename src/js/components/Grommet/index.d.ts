import * as React from "react";

export interface GrommetProps {
  full?: boolean;
  theme?: object;
}

declare const Grommet: React.StatelessComponent<GrommetProps>;

export { Grommet };
