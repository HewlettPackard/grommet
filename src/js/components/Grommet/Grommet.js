import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
