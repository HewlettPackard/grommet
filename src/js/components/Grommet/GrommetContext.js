import React from 'react';

// Track which components mounted onto the DOM are coming from
// Grommet (see FocusedContainer.js)
export const GrommetContext = React.createContext(undefined);
