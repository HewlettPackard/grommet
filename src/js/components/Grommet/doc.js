import { describe, PropTypes } from 'react-desc';

import { getAvailableAtBadge } from '../../utils';

export default (Grommet) => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet'))
    .description('This is the top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`
    );

  DocumentedGrommet.propTypes = {
    dir: PropTypes.oneOf(['rtl', 'ltr']).description(
      `Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.`
    ),
    theme: PropTypes.object.description('Custom styles for Grommet app component.'),
  };

  return DocumentedGrommet;
};
