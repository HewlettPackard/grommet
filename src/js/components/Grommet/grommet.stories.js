import React from 'react';
import { storiesOf } from '@storybook/react';

import { Grommet, Anchor, Box } from 'grommet';
import { Add } from 'grommet-icons';

const customTheme = {
  global: {
    colors: {
      custom: '#cc6633',
    },
  },
};


const Themed = () => (
  <Grommet theme={customTheme}>
    <Box pad='medium'>
      <Anchor icon={<Add />} label='Add' color='custom' />
    </Box>
  </Grommet>
);

storiesOf('Grommet', module)
  .add('Theme', () => <Themed />);
