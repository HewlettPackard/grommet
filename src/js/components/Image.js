// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import classnames from 'classnames';

const CLASS_ROOT = 'image';

const Image = props => {
  let { size, full } = props;
  let classes = classnames(
    CLASS_ROOT,
    props.className,
    {
      [`${CLASS_ROOT}--${size}`]: size,
      [`${CLASS_ROOT}--full`]: typeof full === 'boolean' && full,
      [`${CLASS_ROOT}--full-${full}`]: typeof full === 'string'
    }
  );

  return <img id={props.id} className={classes} src={props.src} />;
};

Image.propTypes = {
  full: PropTypes.oneOf([true, 'horizontal', 'vertical', false]),
  src: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large', 'thumb'])
};

Image.displayName = 'Image';

export default Image;
