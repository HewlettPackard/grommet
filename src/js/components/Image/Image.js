import React, { Component } from 'react';
import { compose } from 'recompose';

import { withTheme } from '../hocs';

import StyledImage from './StyledImage';
import doc from './doc';

class Image extends Component {
  render() {
    return (
      <StyledImage {...this.props} />
    );
  }
}

if (process.env.NODE_ENV !== 'production') {
  doc(Image);
}

export default compose(
  withTheme,
)(Image);
