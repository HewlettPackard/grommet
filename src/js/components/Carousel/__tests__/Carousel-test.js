import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Grommet } from '../../Grommet';
import { Carousel } from '../';
import { Image } from '../../Image';

Enzyme.configure({ adapter: new Adapter() });

test('renders', () => {
  const component = renderer.create(
    <Grommet>
      <Carousel>
        <Image src='//v2.grommet.io/assets/IMG_4245.jpg' />
        <Image src='//v2.grommet.io/assets/IMG_4210.jpg' />
      </Carousel>
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('play renders', (done) => {
  const component = mount(
    <Grommet>
      <Carousel play={1000}>
        <Image src='//v2.grommet.io/assets/IMG_4245.jpg' />
        <Image src='//v2.grommet.io/assets/IMG_4210.jpg' />
      </Carousel>
    </Grommet>
  );
  expect(component.getDOMNode()).toMatchSnapshot();

  // give some time for the carousel to advance
  setTimeout(() => {
    expect(component.getDOMNode()).toMatchSnapshot();
    component.unmount();
    done();
  }, 1300);
});
