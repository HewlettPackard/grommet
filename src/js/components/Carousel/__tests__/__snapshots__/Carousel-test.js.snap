// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`play renders 1`] = `
<div
  class="StyledGrommet-gGRXwz deqpVW"
>
  <div
    class="StyledStack-jqCQsC ewABiA"
  >
    <div
      class="StyledBox-bStriS hLXGcf"
      direction="column"
      overflow="hidden"
      style="position: relative;"
    >
      <div
        class="StyledBox-bStriS gwcVfH"
        direction="column"
      >
        <img
          class="StyledImage-iWJlTu ealRJb"
          src="//v2.grommet.io/assets/IMG_4245.jpg"
        />
      </div>
    </div>
    <div
      class="StyledBox-bStriS hLXGcf"
      direction="column"
      overflow="hidden"
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <div
        class="StyledBox-bStriS eyelgI"
        direction="column"
      >
        <img
          class="StyledImage-iWJlTu ealRJb"
          src="//v2.grommet.io/assets/IMG_4210.jpg"
        />
      </div>
    </div>
    <div
      class="StyledBox-bStriS goqSQA"
      direction="row"
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <div
        class="StyledBox-bStriS TIDVw"
        direction="column"
      >
        <button
          class="StyledButton-gXzblK jZsiZy"
          disabled=""
          type="button"
        >
          <div
            class="StyledBox-bStriS mzwwH"
            direction="column"
          >
            <svg
              class="StyledIcon-gTfjEw iIdMpR"
              width="24px"
              height="24px"
              viewBox="0 0 24 24"
              version="1.1"
              role="img"
              aria-label="Previous"
            >
              <polyline
                fill="none"
                stroke="#000"
                stroke-width="2"
                points="7 2 17 12 7 22"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="StyledBox-bStriS jIVWvO"
        direction="column"
      >
        <div
          class="StyledBox-bStriS cHAEmV"
          direction="row"
        >
          <button
            class="StyledButton-gXzblK etLvGo"
            icon="[object Object]"
            type="button"
          >
            <span
              class="StyledButton__StyledIcon-kzdZcp bhmqMt"
              aria-hidden="true"
            >
              <svg
                class="StyledIcon-gTfjEw cbaxjc"
                width="24px"
                height="24px"
                viewBox="0 0 24 24"
                version="1.1"
                role="img"
                aria-label="Subtract"
                color="brand"
              >
                <path
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                  d="M2,12 L22,12"
                />
              </svg>
            </span>
          </button>
          <button
            class="StyledButton-gXzblK etLvGo"
            icon="[object Object]"
            type="button"
          >
            <span
              class="StyledButton__StyledIcon-kzdZcp bhmqMt"
              aria-hidden="true"
            >
              <svg
                class="StyledIcon-gTfjEw iIdMpR"
                width="24px"
                height="24px"
                viewBox="0 0 24 24"
                version="1.1"
                role="img"
                aria-label="Subtract"
              >
                <path
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                  d="M2,12 L22,12"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="StyledBox-bStriS TIDVw"
        direction="column"
      >
        <button
          class="StyledButton-gXzblK djbfBD"
          type="button"
        >
          <div
            class="StyledBox-bStriS mzwwH"
            direction="column"
          >
            <svg
              class="StyledIcon-gTfjEw iIdMpR"
              width="24px"
              height="24px"
              viewBox="0 0 24 24"
              version="1.1"
              role="img"
              aria-label="Next"
            >
              <polyline
                fill="none"
                stroke="#000"
                stroke-width="2"
                points="7 2 17 12 7 22"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`play renders 2`] = `
<div
  class="StyledGrommet-gGRXwz deqpVW"
>
  <div
    class="StyledStack-jqCQsC ewABiA"
  >
    <div
      class="StyledBox-bStriS hLXGcf"
      direction="column"
      overflow="hidden"
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <div
        class="StyledBox-bStriS eyelgI"
        direction="column"
      >
        <img
          class="StyledImage-iWJlTu ealRJb"
          src="//v2.grommet.io/assets/IMG_4245.jpg"
        />
      </div>
    </div>
    <div
      class="StyledBox-bStriS hLXGcf"
      direction="column"
      overflow="hidden"
      style="position: relative;"
    >
      <div
        class="StyledBox-bStriS itYIan"
        direction="column"
      >
        <img
          class="StyledImage-iWJlTu ealRJb"
          src="//v2.grommet.io/assets/IMG_4210.jpg"
        />
      </div>
    </div>
    <div
      class="StyledBox-bStriS goqSQA"
      direction="row"
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <div
        class="StyledBox-bStriS TIDVw"
        direction="column"
      >
        <button
          class="StyledButton-gXzblK djbfBD"
          type="button"
        >
          <div
            class="StyledBox-bStriS mzwwH"
            direction="column"
          >
            <svg
              class="StyledIcon-gTfjEw iIdMpR"
              width="24px"
              height="24px"
              viewBox="0 0 24 24"
              version="1.1"
              role="img"
              aria-label="Previous"
            >
              <polyline
                fill="none"
                stroke="#000"
                stroke-width="2"
                points="7 2 17 12 7 22"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="StyledBox-bStriS jIVWvO"
        direction="column"
      >
        <div
          class="StyledBox-bStriS cHAEmV"
          direction="row"
        >
          <button
            class="StyledButton-gXzblK etLvGo"
            icon="[object Object]"
            type="button"
          >
            <span
              class="StyledButton__StyledIcon-kzdZcp bhmqMt"
              aria-hidden="true"
            >
              <svg
                class="StyledIcon-gTfjEw iIdMpR"
                width="24px"
                height="24px"
                viewBox="0 0 24 24"
                version="1.1"
                role="img"
                aria-label="Subtract"
              >
                <path
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                  d="M2,12 L22,12"
                />
              </svg>
            </span>
          </button>
          <button
            class="StyledButton-gXzblK etLvGo"
            icon="[object Object]"
            type="button"
          >
            <span
              class="StyledButton__StyledIcon-kzdZcp bhmqMt"
              aria-hidden="true"
            >
              <svg
                class="StyledIcon-gTfjEw cbaxjc"
                width="24px"
                height="24px"
                viewBox="0 0 24 24"
                version="1.1"
                role="img"
                aria-label="Subtract"
                color="brand"
              >
                <path
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                  d="M2,12 L22,12"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="StyledBox-bStriS TIDVw"
        direction="column"
      >
        <button
          class="StyledButton-gXzblK jZsiZy"
          type="button"
          disabled=""
        >
          <div
            class="StyledBox-bStriS mzwwH"
            direction="column"
          >
            <svg
              class="StyledIcon-gTfjEw iIdMpR"
              width="24px"
              height="24px"
              viewBox="0 0 24 24"
              version="1.1"
              role="img"
              aria-label="Next"
            >
              <polyline
                fill="none"
                stroke="#000"
                stroke-width="2"
                points="7 2 17 12 7 22"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders 1`] = `
.c9 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  fill: #666666;
  stroke: #666666;
}

.c9 g {
  fill: inherit;
  stroke: inherit;
}

.c9 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c9 *[stroke*="#"],
.c9 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c9 *[fill-rule],
.c9 *[FILL-RULE],
.c9 *[fill*="#"],
.c9 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c14 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  fill: #666666;
  stroke: #666666;
  fill: #865CD6;
  stroke: #865CD6;
}

.c14 g {
  fill: inherit;
  stroke: inherit;
}

.c14 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c14 *[stroke*="#"],
.c14 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c14 *[fill-rule],
.c14 *[FILL-RULE],
.c14 *[fill*="#"],
.c14 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  overflow: hidden;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  opacity: 1;
  -webkit-animation: gmgyXe 1s 0s forwards;
  animation: gmgyXe 1s 0s forwards;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  padding: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  margin: 0;
  padding: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  padding: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
  padding: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  padding: 0;
}

.c7 {
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  background-color: transparent;
  overflow: visible;
  text-transform: none;
  opacity: 0.3;
  cursor: default;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  width: 100%;
  height: 100%;
  max-width: none;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c12 {
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  background-color: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  padding: 12px;
}

.c15 {
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  background-color: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  width: 100%;
  height: 100%;
  max-width: none;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c15:hover {
  background-color: rgba(221,221,221,0.5);
  color: #000000;
}

.c13 {
  display: inline-block;
}

.c13:first-child:not(:last-child) {
  margin-right: 12px;
}

.c13 > * {
  vertical-align: bottom;
}

.c1 {
  position: relative;
}

@media only screen and (max-width:699px) {
  .c2 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c2 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c3 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c3 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c4 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c4 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c5 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c5 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c6 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c6 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c8 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c8 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c10 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c10 {
    padding: 0;
  }
}

@media only screen and (max-width:699px) {
  .c11 {
    margin: 0;
  }
}

@media only screen and (max-width:699px) {
  .c11 {
    padding: 0;
  }
}

@media only screen and (min-width:700px) {
  .c7 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

@media only screen and (min-width:700px) {
  .c12 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

@media only screen and (min-width:700px) {
  .c15 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onKeyDown={[Function]}
  >
    <div
      aria-label={undefined}
      className="c2"
      direction="column"
      overflow="hidden"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        aria-label={undefined}
        className="c3"
        direction="column"
      >
        <img
          className=""
          src="//v2.grommet.io/assets/IMG_4245.jpg"
        />
      </div>
    </div>
    <div
      aria-label={undefined}
      className="c2"
      direction="column"
      overflow="hidden"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        aria-label={undefined}
        className="c4"
        direction="column"
      >
        <img
          className=""
          src="//v2.grommet.io/assets/IMG_4210.jpg"
        />
      </div>
    </div>
    <div
      aria-label={undefined}
      className="c5"
      direction="row"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        aria-label={undefined}
        className="c6"
        direction="column"
      >
        <button
          aria-label={undefined}
          className="c7"
          disabled={true}
          href={undefined}
          icon={undefined}
          label={undefined}
          onBlur={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <div
            aria-label={undefined}
            className="c8"
            direction="column"
          >
            <svg
              aria-label="Previous"
              className="c9"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                stroke="#000"
                strokeWidth="2"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        aria-label={undefined}
        className="c10"
        direction="column"
      >
        <div
          aria-label={undefined}
          className="c11"
          direction="row"
        >
          <button
            aria-label={undefined}
            className="c12"
            disabled={false}
            href={undefined}
            icon={
              <Subtract
                color="brand"
              />
            }
            label={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="c13"
            >
              <svg
                aria-label="Subtract"
                className="c14"
                color="brand"
                height="24px"
                role="img"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
              >
                <path
                  d="M2,12 L22,12"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label={undefined}
            className="c12"
            disabled={false}
            href={undefined}
            icon={
              <Subtract
                color={undefined}
              />
            }
            label={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="c13"
            >
              <svg
                aria-label="Subtract"
                className="c9"
                color={undefined}
                height="24px"
                role="img"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
              >
                <path
                  d="M2,12 L22,12"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        aria-label={undefined}
        className="c6"
        direction="column"
      >
        <button
          aria-label={undefined}
          className="c15"
          disabled={false}
          href={undefined}
          icon={undefined}
          label={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <div
            aria-label={undefined}
            className="c8"
            direction="column"
          >
            <svg
              aria-label="Next"
              className="c9"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                stroke="#000"
                strokeWidth="2"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
